src/server/api/controllers/admin/EventsController.ts

```diff
@@ -38,7 +35,16 @@ export class EventsController {
     private SponsorService: SponsorService,
     private TeamService: TeamService,
     private UserService: UserService,
-  ) { }
+  ) {
+    // Optionally enable transactions service depending on configuration
+    if (Config.EnableTransactions) {
+      this.TransactionService = Container.get(TransactionService);
+    } else {
+      this.TransactionService = Container.get(MockTransactionService);
+    }
+  }
```

Comment:
@subhankar-panda Oooo conditional dependency injection? Yes please.

Reply:
um no, because we can't rely on dependency injections for this - as we conditionally want to choose the service.

Reply:
Oh I didn't realize those where args to the constructor! so weird that you can annotate those as private lol

Reply:
`private` in this case is syntactic sugar for "Create a private member variable and assign it from this parameter".

```
private TransactionService transactionService;

constructor(TransactionService _transactionService...){
   this.transactionService = _transactionService;
}
```